lib/Jit/LLILCJit.cpp

```diff
@@ -394,3 +407,100 @@ void __cdecl LLILCJit::fatal(int Errnum, ...) {
 void LLILCJit::signalHandler(void *Cookie) {
   // do nothing
 }
+
+void ObjectLoadListener::getDebugInfoForObject(
+    const ObjectFile &Obj, const RuntimeDyld::LoadedObjectInfo &L) {
+  OwningBinary<ObjectFile> DebugObjOwner = L.getObjectForDebug(Obj);
+  const ObjectFile &DebugObj = *DebugObjOwner.getBinary();
+
+  // Get the address of the object image for use as a unique identifier
+  const void *ObjData = DebugObj.getData().data();
+  DIContext *DwarfContext = new DWARFContextInMemory(DebugObj);
```

Comment:
I think that the heap allocation here can be removed. Instead of `DIContext *DwarfContext = new DWARFContextInMemory(DebugObj);`, `DWARFContextInMemory DwarfContext(DebugObj)` should do.